.github/workflows/reusable_build.yml

```diff
@@ -42,47 +50,42 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        target:
-          - SDK: "$NANOS_SDK"
-            name: nanos
-          - SDK: "$NANOX_SDK"
-            name: nanox
-          - SDK: "$NANOSP_SDK"
-            name: nanos2
-          - SDK: "$STAX_SDK"
-            name: stax
+        device: ${{ fromJSON(inputs.run_for_devices) }}
     runs-on: ubuntu-latest
     container:
       image: ghcr.io/ledgerhq/ledger-app-builder/${{ inputs.builder }}:latest
 
     steps:
+      - name: Processing the BOLOS_SDK and DEVICE_NAME values
+        run: |
+          echo "DEVICE_NAME=$(echo ${{ matrix.device }} | tr nanosp nanos2)" >> $GITHUB_ENV
+          eval "var=\$$(echo ${{ matrix.device }} | tr [:lower:] [:upper:])_SDK" && echo "BOLOS_SDK=$var" >> $GITHUB_ENV
+
```

Comment:
With the other suggestion, no need to use the github env anymore as both variables are only used in a single subshell so you can use `&&`